pruntime/enclave/src/contracts/diem.rs

```diff
@@ -0,0 +1,270 @@
+use core::convert::TryFrom;
+use diem_types::epoch_change::EpochChangeProof;
+use diem_types::ledger_info::LedgerInfoWithSignatures;
+use diem_types::trusted_state::{TrustedState, TrustedStateChange};
+use diem_types::proof::{
+    AccountStateProof, TransactionInfoWithProof,
+    TransactionAccumulatorProof, SparseMerkleProof,
+};
+use diem_types::account_address::AccountAddress;
+use diem_types::account_state_blob::AccountStateBlob;
+use diem_types::transaction::TransactionInfo;
+use diem_crypto::hash::CryptoHash;
+use diem_types::transaction::{Transaction, SignedTransaction, TransactionPayload};
+use move_core_types::transaction_argument::TransactionArgument;
+
+#[derive(Clone, Serialize, Deserialize, Debug, PartialEq)]
+pub struct Amount {
+    pub amount: u64,
+    pub currency: String,
+}
+
+#[derive(Debug, Clone, Serialize, Deserialize)]
+pub struct AccountInfo {
+    pub address: AccountAddress,
+    pub authentication_key: Option<Vec<u8>>,
+    pub sequence_number: u64,
+    pub sent_events_key: String,
+    pub received_events_key: String,
+    pub balances: Vec<Amount>,
+}
+
+#[derive(Debug, Serialize, Deserialize, Clone)]
+pub struct TransactionWithProof {
+    transaction_bytes: Vec<u8>,
+
+    epoch_change_proof: EpochChangeProof,
+    ledger_info_with_signatures: LedgerInfoWithSignatures,
+
+    ledger_info_to_transaction_info_proof: TransactionAccumulatorProof,
+    transaction_info: TransactionInfo,
+    transaction_info_to_account_proof: SparseMerkleProof,
+    account_state_blob: AccountStateBlob,
+
+    version: u64,
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Error {
+    NotAuthorized,
+    Other(String),
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Command {
+}
+
+#[derive(Serialize, Deserialize, Debug, Clone)]
+pub enum Request {
+    AccountData { account_data_b64: String },
+    VerifyTransaction { account_address: String, transaction_with_proof_b64: String }
+}
+
+#[derive(Serialize, Deserialize, Debug)]
+pub enum Response {
+    AccountData { size: u32 },
+    VerifyTransaction { total: u32, verified: bool },
+    Error(Error)
+}
+
+#[derive(Serialize, Deserialize)]
+pub struct Diem {
+    accounts: Vec<AccountInfo>,
+    transactions: BTreeMap<String, Vec<Transaction>>, //address => Transaction
+    proofs: BTreeMap<String, TransactionWithProof>, //Hash => Proof
+}
+
+impl Diem {
+    pub fn new() -> Self {
+        Diem {
+            accounts: Vec::new(),
+            transactions: BTreeMap::<String, Vec<Transaction>>::new(),
+            proofs: BTreeMap::<String, TransactionWithProof>::new(),
+        }
+    }
+}
+
+impl contracts::Contract<Command, Request, Response> for Diem {
+    fn id(&self) -> contracts::ContractId { contracts::DIEM }
+
+    fn handle_command(&mut self, _origin: &chain::AccountId, _txref: &TxRef, _cmd: Command) -> TransactionStatus {
+        TransactionStatus::Ok
+    }
+
+    fn handle_query(&mut self, origin: Option<&chain::AccountId>, req: Request) -> Response {
```

Comment:
Nice work in general! Next step, we should break it down to a few functions to make the code clearer